stdlib/sys.mli

```diff
@@ -354,6 +354,18 @@ val runtime_warnings_enabled: unit -> bool
 
     @since 4.03.0 *)
 
+val enable_float_array_alloc_warning: bool -> int -> unit
+(** Control whether the OCaml runtime system can emit warnings when values
+    of type [float array] arrays are allocated. The second argument specifies
+    the size of the printed function callstack.
+
```

Comment:
Is the second argument just ignored when the first argument is `false`?  This suggests changing the function to take an `int option` argument, no?  And, to be symmetric and allow overriding the setting for a limited dynamic scope, the getter function would also return a value of the same type.

Reply:
(Or just use an `int` as on the C side, and document that <= 0 means the warning is disabled.)

Reply:
I argued against this, as it seemed cleaner to separate the two arguments, but sure, why not...

Reply:
Just curiously, are there any option to deconstruct an option value in the C code ? (otherwise just a small wrapper around the C call suffices of course)

Reply:
> I argued against this, as it seemed cleaner to separate the two arguments, but sure, why not...

The way I see it is that the two arguments are really related: one is useless for half the possible values of the other one :-)

If we think about the future addition of other similar warnings, with their own enabling/disabling functions, it could make sense to have just one global setting for the printed stack depth (used for all warnings).

Reply:
> > I argued against this, as it seemed cleaner to separate the two arguments, but sure, why not...
> 
> The way I see it is that the two arguments are really related: one is useless for half the possible values of the other one :-)

Yes, I agree that going with an option type is better.

Reply:
> Just curiously, are there any option to deconstruct an option value in the C code ? (otherwise just a small wrapper around the C call suffices of course)

Yes, you check if you are in the `None` case with `Is_block` and if you are in the `Some` case you can extract the payload with `Field(v, 0)`.

Reply:
Nice, thanks for the tip. In terms of readability though, is it better to do this in the runtime part or in the stdlib part ?

Reply:
No strong opinion either way. Note that if you switch to an option, you will need to handle the case `Some n` when `n <= 0` and raise `invalid_arg` in that case.

Reply:
Note: we now have option-handling macros in mlvalues.h:

```
#define Val_none Val_int(0)
#define Some_val(v) Field(v, 0)
#define Tag_some 0
#define Is_none(v) ((v) == Val_none)
#define Is_some(v) Is_block(v)
```

(I think they were implemented by @nojb?)

Reply:
Indeed, ouch! :)